src/naive/period.rs

```diff
@@ -0,0 +1,279 @@
+use naive::NaiveDateTime;
+use oldtime::Duration;
+
+use std::cmp::{min, max};
+
+/// A period of time between two ISO 8601 dates/times ([NaiveDateTime](NaiveDateTime)s).
+/// This is similar to a [`Duration`](Duration), except that it has a fixed start
+/// date/time (and thus a fixed end date/time), allowing clients to determine if specific
+/// segments of time intersect.
+///
+/// # Notes
+/// This period is not reliant on time zones. For the time being, offsets aren't considered at all.
+/// As such, if you need to interset two time periods that are of differing time zones, you're out
+/// of luck (patches accepted, though).
+#[derive(PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Debug)]
+pub struct NaivePeriod {
+    /// The date at which the period begins.
+    pub start: NaiveDateTime,
+
+    /// The date at which the period ends. This is inclusive, meaning that the period runs up to
+    /// and including this date/time.
+    pub end: NaiveDateTime
+}
+
+impl NaivePeriod {
+    /// Create a new `NaivePeriod` from two [`NaiveDateTime`](NaiveDateTime) objects.
+    ///
+    /// # Arguments
+    /// - `start`: A `NaiveDateTime` representing when the `NaivePeriod` will start.
+    /// - `end`: A `NaiveDateTime` representing when the `NaivePeriod` will end. Note that this
+    ///   date/time will be included in the period itself.
+    ///
+    /// # Returns
+    /// - A `NaivePeriod` object having the specified start and end `NaiveDateTime`s.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::{Duration, NaiveDate, NaiveDateTime, NaivePeriod, NaiveTime};
+    /// let start = NaiveDateTime::new(NaiveDate::from_ymd(2020, 1, 1), NaiveTime::from_hms(0, 0, 0));
+    /// let end = NaiveDateTime::new(NaiveDate::from_ymd(2021, 1, 1), NaiveTime::from_hms(0, 0, 0));
+    ///
+    /// let np1 = NaivePeriod::new(start, end);
+    ///
+    /// assert_eq!(np1.duration(), Duration::days(366));
+    /// ```
+    #[inline]
+    pub fn new(start: NaiveDateTime, end: NaiveDateTime) -> Self {
+        NaivePeriod { start: start, end: end }
```

Comment:
Is it reasonable for us to verify that `start < end` or `start <= end`? Maybe rename to `left, right` and just put them in the right order?